packages/headless/src/controllers/result-list/headless-result-list.ts

```diff
@@ -51,6 +53,22 @@ export function buildResultList(
     dispatch(registerFieldsToInclude(options.fieldsToInclude));
   }
 
+  const triggerFetchMore = (shouldKeepFetchingResults?: () => boolean) => {
```

Comment:
Why is `shouldKeepFetchingResults` optional? Is it not to force the user to give a callback function and, int that case, only one batch of result will be fetched?